### DISCUSSION TITLE
PX4 firmware update returns  warining for missing parameters

### DISCUSSION THREAD

**Discussion by maxhofidrone** [ctu-mrs/mrs_uav_system#186] [Q&A] [answered]
Hi,
I installed the MRS System on a plain Ubuntu 20. I tried to find your custom px4 firmware but did not find it (only a gazebo version). I then cloned the firmware as described here: 
https://ctu-mrs.github.io/docs/hardware/px4_firmware.html
and installed all the necessary toolchain from here:
https://docs.px4.io/main/en/dev_setup/dev_env_linux_ubuntu.html

I built the  px4_fmu-v5_default version and updated the firmware with QGC. After the update I got following error message in QGC:
![image](https://github.com/ctu-mrs/mrs_uav_system/assets/55589716/faba454a-3763-4e0b-830d-885b0c00eb07)

These parameters were available in the old firmware. I attached the parameter file of the old firmware. Is this error/warning to be expected? Or did I install the wrong Firmware?
[param_px4.params.txt](https://github.com/user-attachments/files/16124623/param_px4.params.txt)

Thank you very much for any help in advance!

**Comment by spurnvoj** [comment]
Hi @maxhofidrone, If you installed the system natively using PPA as suggested in https://github.com/ctu-mrs/mrs_uav_system?tab=readme-ov-file#native-installation. The px4 binary is installed in your system under a package named "mrs-noetic-px4".

**Reply by spurnvoj** [reply]
Of course, this version can only be used for SITL with a gazebo. If you want to get the binary for your Pixhawk autopilot that is compatible with the current version of the MRS system, please let us know which version you have, and we can send it to you. @DanHert , please help with this if needed.

**Reply by maxhofidrone** [reply]
Hi @spurnvoj, 
yes I installed it using the stable PPA with the help of this [script](https://github.com/fly4future/install_script). I also updated the system with the latest PPA today. The FCU is the Pixhawk4. Would be great if you could sent me the correct binary!
Thank you very much in advance!

**Comment by DanHert** [comment, marked as answer]
Hi @maxhofidrone, the repository with the correct firmware fork is here - https://github.com/ctu-mrs/px4_firmware

I am also attaching a binary for fmu-V5:
[fw.zip](https://github.com/user-attachments/files/16280789/fw.zip)

**Reply by maxhofidrone** [reply]
Thank you very much!

**Comment by MJavadZallaghi** [comment]
Hi @DanHert and @spurnvoj. I will use Pixhawk 6x as flight controller and I'm considering using MRS System with it.
I saw [https://github.com/ctu-mrs/px4_firmware/releases/tag/v1.14.0](https://github.com/ctu-mrs/px4_firmware/releases/tag/v1.14.0) that means you have provided version 1.14 for PX4. To be sure, is this version (1.14) compatible with Pixhawk 6x I'm considering for my aerial robot?
Thank you!

**Reply by DanHert** [reply]
Hello, we have tested the system with even the newer release - 1.15.2. Everything works without any modifications. The only thing that works differently with 1.15.2 is that if you are flying in the Offboard mode and you do any flight mode change (like Altitude->Position) the UAV will disengage from the Offboard mode. Other than that, everything else works fine.
